src/Server/ServerPurchases.ts

```diff
@@ -17,46 +17,50 @@ import { workerScripts } from "../Netscript/WorkerScripts";
 // Returns Infinity for invalid 'ram' arguments
 /**
  * @param ram Amount of RAM on purchased server (GB)
+ * @param cores Amount of cores on purchased Server 
  * @returns Cost of purchasing the given server. Returns infinity for invalid arguments
  */
-export function getPurchaseServerCost(ram: number): number {
+export function getPurchaseServerCost(ram: number, cores :number): number {
   const sanitizedRam = Math.round(ram);
   if (isNaN(sanitizedRam) || !isPowerOfTwo(sanitizedRam) || !(Math.sign(sanitizedRam) === 1)) {
     return Infinity;
   }
-
   if (sanitizedRam > getPurchaseServerMaxRam()) {
     return Infinity;
   }
 
   const upg = Math.max(0, Math.log(sanitizedRam) / Math.log(2) - 6);
-
```

Comment:
This would be a good place to verify that `cores` is a positive integer.

Reply:
oh thats actually what i meant with "layer 3" 
its only called once from outside the layer chain i described and that are the tech stores where i set it to 1
layer 1 and 2 should do the validation and what every gets passed into that should be valid imo

Reply:
Ah, in that case the layer 1 checks could use `helpers.positiveInteger` instead of `helpers.number`